### DISCUSSION TITLE
Maximum number of volumes

### DISCUSSION THREAD

**Discussion by SmsS4** [seaweedfs/seaweedfs#5164] [Q&A]
I'm using Seaweedfs s3 for our production.
And we have small and big customers.
Seaweed use `volumeSizeLimitMB` * 7 for each s3 bucket so I have to set `volumeSizeLimitMB` to a small number (for example 1GB)
If I want to store 10 petabytes of data with multiple volume servers with 10 terabytes disk I will have 1000 volume servers and 10485760 volumes.
Does seaweedfs works fast with these number of volumes?
What is maximum number of volumes seaweedfs can handle?

Thanks.

**Comment by amir-latifi** [comment]
Hi
This is my question too. Could you please answer it? @chrislusf 
Has anyone  had such an experience with seaweed? Does the performance remain normal?

**Reply by kmlebedev** [reply]
> Does seaweedfs works fast with these number of volumes?

This will not have any effect on speed, since the topology of revolutions by server is in the background. But this can significantly affect memory consumption and CPC utilization.
It is better to use the largest possible revolution, which corresponds to a reasonable transfer rate of one volume of file to the backup server.
Or simply turn off prelocation if you realize that you will have a larger number of partially filled volumes.
